pkg/cgroups/cgroups_linux.go

```diff
@@ -45,6 +49,51 @@ func mountCgroup() error {
 	return nil
 }
 
+// cgrpSanityCheck performs a sanity check on the mounted cgroup fs and issues
+// a warning if something seems wrong.
+//
+// It does this by recursively checking the number of pids in "cgroup.procs". If they are 2
+// or less, it is likely that the cilium pod is the only container in this
```

Comment:
Why 2 is still bad?

Reply:
It's just a heuristic. I thought to count the two containers running on the pod: the init container and the agent container.

Reply:
I think init container is started before cilium-agent, and it doesn't run after cilium-agent has been started.

Reply:
Ah, I see. Should I change this to 1 then you think?

Reply:
Pinging @aanm to confirm my hypothesis ^^.

Reply:
>I think init container is started before cilium-agent, and it doesn't run after cilium-agent has been started.

* Init containers always run to completion.
* Each init container must complete successfully before the next one starts.

https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#understanding-init-containers

Reply:
Thanks. Then we need to change to 1. Just the question - have we ever observed a case where cgroup.procs=0?

Reply:
OK, will change it to 1.

> have we ever observed a case where cgroup.procs=0

For cgroup that the cilium agent is mounting I have not. In general, yes.

Reply:
I wonder whether it's always guaranteed that cilium-agent will run in the `cgroupRoot`. If yes, then I guess this sanity check might be redundant.

Reply:
Updated to use check for <=1 processes under the cgroup.